File: 프로그래머스/0/120852. 소인수분해/소인수분해.py
def solution(n):
    answer = []
    answerset = set()
    i = 2
    while n>1 :
        if n%i == 0 :
            answerset.add(i)
            n/= i
        else :
            i+=1 
            
    answer = list(answerset)
    answer.sort()
    
    return answer

File: 프로그래머스/0/120852. 소인수분해/소인수분해.java
import java.util.*;

class Solution {
    public int[] solution(int n) {
        Set<Integer> factors = new HashSet<>();

        for (int i = 2; i <= n; i++) {
            while (n % i == 0) {
                factors.add(i);
                n /= i;
            }
        }

        // 중복 제거된 소인수들을 정렬하여 배열에 저장
        int[] answer = new int[factors.size()];
        int index = 0;
        for (int factor : factors) {
            answer[index++] = factor;
        }
        Arrays.sort(answer);

        return answer;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int n = 24;
        int[] result = solution.solution(n);
        System.out.println(Arrays.toString(result)); // [2, 3]
    }
}
